nntrainer/src/tensor.cpp

```diff
@@ -76,24 +76,19 @@ static auto rng = [] {
   return rng;
 }();
 
-Tensor::Tensor(const TensorDim &d, const float *buf) :
-  dim(d),
-  strides{{1, 2, 3}},
-  is_contiguous(true),
-  data(d.getDataLen() == 0
-         ? nullptr
-         : std::shared_ptr<float>(new float[d.getDataLen()],
-                                  std::default_delete<float[]>())) {
-  if (d.getDataLen() == 0) {
-    return;
-  }
-
-  // todo: initialize appropriate strides
-  if (buf != nullptr) {
-    float *data = getData();
-    unsigned int len = length();
-
-    scopy(len, buf, 1, data, 1);
+Tensor::Tensor(const TensorDim &d, const float *buf) : Tensor() {
```

Comment:
IMO, `dim = d` should be done even if it is empty. It might cause a problem when tensordim starts to have any meta information.

Reply:
Good catch. Updated.

Reply:
This creates an issue with the bn_layer.
bn creates weights without gradient but passes it to the optimizer.
The optimizer creates its own variables with that empty gradient which throws error of creating tensor with 0 dimensions.
How about I disable the bn unit test and you can check this while you implementing it?

Reply:
AFAIK, this should not affect as bn_layer does not pass weights that does not have grads. I'll check it through anyway. Thanks.

Reply:
@zhoonit #529 resolves this issue.